### DISCUSSION TITLE
MS Access does not work with VS2022 64 BIT

### DISCUSSION THREAD

**Discussion by AndreasWaning** [linq2db/linq2db#3345] [Q&A] [answered]
With VS 2022 Linq2db is not executing T4 for a Access Database. Because VS 2022 ist 64 bit.

Is there baybe a solution avaible ?

Thanks Andreas

**Comment by MaceWindu** [comment]
You need:
1. x64-bit oledb or odbc provider installed

to see providers you can use:
For `ODBC` `Get-OdbcDriver | Select Name, Platform` powershell command:
![image](https://user-images.githubusercontent.com/1099238/141652140-0db53977-7c8c-4e8f-98c0-535011c48616.png)
For `OleDb` you can use `[System.Data.OleDb.OleDbEnumerator]::GetRootEnumerator() | % { $_.GetValue(0) + " => " + $_.GetValue(2) }` command, but it doesn't show bitness
![image](https://user-images.githubusercontent.com/1099238/141652489-21f3280e-32fa-4521-bdc2-6b6bd03098a2.png)


Some examples you could try:
- "modern" `ODBC` connection string: `"Driver={Microsoft Access Driver (*.mdb, *.accdb)};Dbq=<PATH_TO_YOUR_DB>;ExtendedAnsiSQL=1"`
- "less modern" `ODBC` connection string (no `accdb` support): `"Driver={Microsoft Access Driver (*.mdb)};Dbq=<PATH_TO_YOUR_DB>;ExtendedAnsiSQL=1"`
- `OLEDB` connection string (v12): `"Provider=Microsoft.ACE.OLEDB.12.0;Data Source=<PATH_TO_YOUR_DB>;"`
- `OLEDB` connection string (v15): `"Provider=Microsoft.ACE.OLEDB.15.0;Data Source=<PATH_TO_YOUR_DB>;"`

Note that it will not work with `Microsoft.Jet.OLEDB.4.0` provider, as Microsoft never released x64 version of `Jet`

**Reply by AndreasWaning** [reply]
Thank you for helping. I will Try it.

**Comment by MaceWindu** [comment, marked as answer]
New scaffold tool (#3098) will have `architecture` switch to handle required process bitness for unmanaged providers
